### PR TITLE
Adds core tests job while including the mu-plugins

### PR DESCRIPTION
**Pull request by pschoffer**
## Description

We wanted to be able to test mu-plugins against core WordPress tests.

## Steps to Test

Check the CircleCI jobs linked to this PR.

### CODE REVIEW

**Comment by pschoffer** [comment]
Well this is not the most elegant way of solving it, but it turns out that out of the box mu-plugins will make several tens of tests to fail or error.

So in here I just excluded those problematic tests from execution. We still get most of the core-tests and we can figure out later if and when we would like to fix the other tests. 

The fails differ a lot. Sometimes it is a wrong text in some email, sometimes two numbers differ, some fail due to WP_filesystem, and many more.

File: .circleci/config.yml [hunk 1]
```diff
@@ -123,6 +187,16 @@ jobs:
       - image: wordpressdevelop/php:7.3-fpm
       - image: *db_image
 
+  php73-core-tests-mu-plugins:
+    <<: *php_core_job
+    environment:
+      - INCLUDE_MU_TESTS: "1"
+      - WP_MULTISITE: "0"
```
Comment (nickdaugherty) [inline .circleci/config.yml:194]: We should run a multisite version of this too.

File: .circleci/config.yml [hunk 2]
```diff
@@ -96,6 +97,69 @@ job-references:
             npm ci
             npm run grunt build
             export WP_TESTS_DIR=<< pipeline.parameters.wp_core_develop_dir >>/tests/phpunit
+            if [[ "$INCLUDE_MU_TESTS" == "1" ]]; then
+              sed -i "s|bootstrap=\"tests/phpunit/includes/bootstrap.php\"|bootstrap=\"$HOME/project/tests/bootstrap.php\"|" phpunit.xml.dist
```
Comment (nickdaugherty) [inline .circleci/config.yml:101]: Does replacing the core bootstrap file cause the later test failures? Our bootstrap file doesn't necessarily mean that all of mu-plugins has been loaded, just the tested stuff, so it's not a 1:1 representation of what actually runs in production.

What we really need to do is make sure the mu-plugin loading code from core's wp-settings.php is running during these tests. Maybe we can modify the wp-settings.php from the tests to include core's mu-plugin loading code?

**Review by nickdaugherty — CHANGES_REQUESTED** [review]
Nice - I left a few thoughts on the bootstrapping and running this for multisites.

**Comment by pschoffer** [comment]
I have done some investigation into the first from 2xx failing tests. It turns out that our `mu-plugins`:

/home/pavel/git/automattic/vip-go-mu-dev/mu-plugins/vip-mail.php:27 - hooks into `wp_mail_from` and changes the email to `donotreply@wordpress.com`, which breaks the assertion of several tests that set via a different filter (`user_request_action_email_headers`) the from address to be `tester@example.com` 

e.g.: 
/home/pavel/svn/wordpress/5.5.1/tests/phpunit/tests/user/wpSendUserRequest.php:247

I am not sure how to handle it. I guess we could change the core test to remove those kind of filters in that test, but that seems wrong, as the change is triggered by some unrelated plugin code to the core. Also, considering how many failing tests are there and how long it took me to track the reason for failing of this one, I don't think that is sustainable:)

I would suggest skipping all the failing tests as before. Even though we would not run the full test suite, we still get most of the core tests in that way increasing our certainty that we didn't break core I think.

**Comment by nickdaugherty** [comment]
Alright - that email address test is a good example of tests that should be fine to skip - making them pass isn't worth the effort.

Have you looked through all the failing tests @pschoffer? Since each test that we skip represents a gap in coverage, I just want to be deliberate in our decision to skip a given test (we should only skip it if we're confident it doesn't pose a risk and it's not really fixable).

File: .circleci/config.yml (hunk 1 above)
Comment (pschoffer) [reply .circleci/config.yml:144]: Done

File: bin/utils.sh [hunk 3]
```diff
@@ -27,4 +27,90 @@ install_db() {
 	mysqladmin create $DB_NAME --user="$DB_USER" --password="$DB_PASS"$EXTRA
 }
 
-export -f install_db
\ No newline at end of file
+exclude_core_tests() {
+	local FILE=$1
+	local PREFIX=$2 || ""
+	local TO_EXCLUDE=(
+		"tests/term/wpDeleteTerm.php"
```
Comment (brandon-m-skinner) [inline bin/utils.sh:34]:
```suggestion
	"tests/term/wpDeleteTerm.php"
```

File: bin/utils.sh [hunk 4]
```diff
@@ -27,4 +27,90 @@ install_db() {
...
+        "tests/xmlrpc/wp/deleteTerm.php"
+        "tests/xmlrpc/wp/deletePost.php"
+        "tests/xmlrpc/mw/newPost.php"
+        "tests/xmlrpc/mw/getRecentPosts.php"
+        "tests/xmlrpc/mw/getPost.php"
+        "tests/xmlrpc/mw/editPost.php"
+        "tests/xmlrpc/mt/getRecentPostTitles.php"
+        "tests/rest-api/rest-users-controller.php"
+        "tests/filesystem/findFolder.php"
+        "tests/admin/includesPlugin.php"
```
Comment (brandon-m-skinner) [inline bin/utils.sh:107]:
```suggestion
		"tests/term/wpDeleteTerm.php"
        	"tests/term/query.php"
        	"tests/term/getTerms.php"
        	"tests/xmlrpc/basic.php"
        	"tests/user/wpSendUserRequest.php"
        	"tests/user/capabilities.php"
        	"tests/term/wpGetObjectTerms.php"
        	"tests/term/getTheTerms.php"
        	"tests/term/getTerm.php"
        	"tests/term/cache.php"
        	"tests/term.php"
        	"tests/taxonomy.php"
        	"tests/sitemaps/sitemaps.php"
        	"tests/sitemaps/functions.php"
        	"tests/rest-api/rest-schema-setup.php"
        	"tests/rest-api/rest-posts-controller.php"
        	"tests/rest-api/rest-attachments-controller.php"
        	"tests/query/search.php"
        	"tests/query/isTerm.php"
        	"tests/privacy/wpPrivacySendRequestConfirmationNotification.php"
        	"tests/privacy/wpPrivacySendPersonalDataExportEmail.php"
        	"tests/privacy/wpPrivacySendErasureFulfillmentNotification.php"
        	"tests/post/getLastPostModified.php"
        	"tests/option/updateOption.php"
        	"tests/mail.php"
        	"tests/import/postmeta.php"
        	"tests/import/parser.php"
        	"tests/import/import.php"
        	"tests/filesystem/base.php"
        	"tests/feed/rss2.php"
        	"tests/dependencies/scripts.php"
        	"tests/date/getFeedBuildDate.php"
        	"tests/customize/custom-css-setting.php"
        	"tests/comment/wpCountComments.php"
        	"tests/canonical/sitemaps.php"
        	"tests/admin/includesScreen.php"
        	"tests/filesystem/base.php"
        	"tests/feed/atom.php"
        	"tests/adminbar.php"
        	"tests/xmlrpc/wp/restoreRevision.php"
        	"tests/xmlrpc/wp/newTerm.php"
        	"tests/xmlrpc/wp/newPost.php"
        	"tests/xmlrpc/wp/getUsers.php"
        	"tests/xmlrpc/wp/getUser.php"
        	"tests/xmlrpc/wp/getTerms.php"
        	"tests/xmlrpc/wp/getTerm.php"
        	"tests/xmlrpc/wp/getTaxonomy.php"
        	"tests/xmlrpc/wp/getTaxonomies.php"
        	"tests/xmlrpc/wp/getRevisions.php"
        	"tests/xmlrpc/wp/getProfile.php"
        	"tests/xmlrpc/wp/getPosts.php"
        	"tests/xmlrpc/wp/getPostTypes.php"
        	"tests/xmlrpc/wp/getPostType.php"
        	"tests/xmlrpc/wp/getPost.php"
        	"tests/xmlrpc/wp/getPages.php"
        	"tests/xmlrpc/wp/getPageList.php"
        	"tests/xmlrpc/wp/getPage.php"
        	"tests/xmlrpc/wp/getOptions.php"
        	"tests/xmlrpc/wp/getMediaItem.php"
        	"tests/xmlrpc/wp/getComments.php"
        	"tests/xmlrpc/wp/getComment.php"
        	"tests/xmlrpc/wp/editTerm.php"
        	"tests/xmlrpc/wp/editProfile.php"
        	"tests/xmlrpc/wp/editPost.php"
        	"tests/xmlrpc/wp/deleteTerm.php"
        	"tests/xmlrpc/wp/deletePost.php"
        	"tests/xmlrpc/mw/newPost.php"
        	"tests/xmlrpc/mw/getRecentPosts.php"
        	"tests/xmlrpc/mw/getPost.php"
        	"tests/xmlrpc/mw/editPost.php"
        	"tests/xmlrpc/mt/getRecentPostTitles.php"
        	"tests/rest-api/rest-users-controller.php"
        	"tests/filesystem/findFolder.php"
        	"tests/admin/includesPlugin.php"
```

Comment (brandon-m-skinner) [inline bin/utils.sh:107]:
```suggestion
		"tests/term/wpDeleteTerm.php"
        	"tests/term/query.php"
        	"tests/term/getTerms.php"
        	"tests/xmlrpc/basic.php"
        	"tests/user/wpSendUserRequest.php"
        	"tests/user/capabilities.php"
        	"tests/term/wpGetObjectTerms.php"
        	"tests/term/getTheTerms.php"
        	"tests/term/getTerm.php"
        	"tests/term/cache.php"
        	"tests/term.php"
        	"tests/taxonomy.php"
        	"tests/sitemaps/sitemaps.php"
        	"tests/sitemaps/functions.php"
        	"tests/rest-api/rest-schema-setup.php"
        	"tests/rest-api/rest-posts-controller.php"
        	"tests/rest-api/rest-attachments-controller.php"
        	"tests/query/search.php"
        	"tests/query/isTerm.php"
        	"tests/privacy/wpPrivacySendRequestConfirmationNotification.php"
        	"tests/privacy/wpPrivacySendPersonalDataExportEmail.php"
        	"tests/privacy/wpPrivacySendErasureFulfillmentNotification.php"
        	"tests/post/getLastPostModified.php"
        	"tests/option/updateOption.php"
        	"tests/mail.php"
        	"tests/import/postmeta.php"
        	"tests/import/parser.php"
        	"tests/import/import.php"
        	"tests/filesystem/base.php"
        	"tests/feed/rss2.php"
        	"tests/dependencies/scripts.php"
        	"tests/date/getFeedBuildDate.php"
        	"tests/customize/custom-css-setting.php"
        	"tests/comment/wpCountComments.php"
        	"tests/canonical/sitemaps.php"
        	"tests/admin/includesScreen.php"
        	"tests/filesystem/base.php"
        	"tests/feed/atom.php"
        	"tests/adminbar.php"
        	"tests/xmlrpc/wp/restoreRevision.php"
        	"tests/xmlrpc/wp/newTerm.php"
        	"tests/xmlrpc/wp/newPost.php"
        	"tests/xmlrpc/wp/getUsers.php"
        	"tests/xmlrpc/wp/getUser.php"
        	"tests/xmlrpc/wp/getTerms.php"
        	"tests/xmlrpc/wp/getTerm.php"
        	"tests/xmlrpc/wp/getTaxonomy.php"
        	"tests/xmlrpc/wp/getTaxonomies.php"
        	"tests/xmlrpc/wp/getRevisions.php"
        	"tests/xmlrpc/wp/getProfile.php"
        	"tests/xmlrpc/wp/getPosts.php"
        	"tests/xmlrpc/wp/getPostTypes.php"
        	"tests/xmlrpc/wp/getPostType.php"
        	"tests/xmlrpc/wp/getPost.php"
        	"tests/xmlrpc/wp/getPages.php"
        	"tests/xmlrpc/wp/getPageList.php"
        	"tests/xmlrpc/wp/getPage.php"
        	"tests/xmlrpc/wp/getOptions.php"
        	"tests/xmlrpc/wp/getMediaItem.php"
        	"tests/xmlrpc/wp/getComments.php"
        	"tests/xmlrpc/wp/getComment.php"
        	"tests/xmlrpc/wp/editTerm.php"
        	"tests/xmlrpc/wp/editProfile.php"
        	"tests/xmlrpc/wp/editPost.php"
        	"tests/xmlrpc/wp/deleteTerm.php"
        	"tests/xmlrpc/wp/deletePost.php"
        	"tests/xmlrpc/mw/newPost.php"
        	"tests/xmlrpc/mw/getRecentPosts.php"
        	"tests/xmlrpc/mw/getPost.php"
        	"tests/xmlrpc/mw/editPost.php"
        	"tests/xmlrpc/mt/getRecentPostTitles.php"
        	"tests/rest-api/rest-users-controller.php"
        	"tests/filesystem/findFolder.php"
        	"tests/admin/includesPlugin.php"
```

Comment (brandon-m-skinner) [inline bin/utils.sh:107]:
```suggestion
		"tests/term/wpDeleteTerm.php"
        	"tests/term/query.php"
        	"tests/term/getTerms.php"
        	"tests/xmlrpc/basic.php"
        	"tests/user/wpSendUserRequest.php"
        	"tests/user/capabilities.php"
        	"tests/term/wpGetObjectTerms.php"
        	"tests/term/getTheTerms.php"
        	"tests/term/getTerm.php"
        	"tests/term/cache.php"
        	"tests/term.php"
        	"tests/taxonomy.php"
        	"tests/sitemaps/sitemaps.php"
        	"tests/sitemaps/functions.php"
        	"tests/rest-api/rest-schema-setup.php"
        	"tests/rest-api/rest-posts-controller.php"
        	"tests/rest-api/rest-attachments-controller.php"
        	"tests/query/search.php"
        	"tests/query/isTerm.php"
        	"tests/privacy/wpPrivacySendRequestConfirmationNotification.php"
        	"tests/privacy/wpPrivacySendPersonalDataExportEmail.php"
        	"tests/privacy/wpPrivacySendErasureFulfillmentNotification.php"
        	"tests/post/getLastPostModified.php"
        	"tests/option/updateOption.php"
        	"tests/mail.php"
        	"tests/import/postmeta.php"
        	"tests/import/parser.php"
        	"tests/import/import.php"
        	"tests/filesystem/base.php"
        	"tests/feed/rss2.php"
        	"tests/dependencies/scripts.php"
        	"tests/date/getFeedBuildDate.php"
        	"tests/customize/custom-css-setting.php"
        	"tests/comment/wpCountComments.php"
        	"tests/canonical/sitemaps.php"
        	"tests/admin/includesScreen.php"
        	"tests/filesystem/base.php"
        	"tests/feed/atom.php"
        	"tests/adminbar.php"
        	"tests/xmlrpc/wp/restoreRevision.php"
        	"tests/xmlrpc/wp/newTerm.php"
        	"tests/xmlrpc/wp/newPost.php"
        	"tests/xmlrpc/wp/getUsers.php"
        	"tests/xmlrpc/wp/getUser.php"
        	"tests/xmlrpc/wp/getTerms.php"
        	"tests/xmlrpc/wp/getTerm.php"
        	"tests/xmlrpc/wp/getTaxonomy.php"
        	"tests/xmlrpc/wp/getTaxonomies.php"
        	"tests/xmlrpc/wp/getRevisions.php"
        	"tests/xmlrpc/wp/getProfile.php"
        	"tests/xmlrpc/wp/getPosts.php"
        	"tests/xmlrpc/wp/getPostTypes.php"
        	"tests/xmlrpc/wp/getPostType.php"
        	"tests/xmlrpc/wp/getPost.php"
        	"tests/xmlrpc/wp/getPages.php"
        	"tests/xmlrpc/wp/getPageList.php"
        	"tests/xmlrpc/wp/getPage.php"
        	"tests/xmlrpc/wp/getOptions.php"
        	"tests/xmlrpc/wp/getMediaItem.php"
        	"tests/xmlrpc/wp/getComments.php"
        	"tests/xmlrpc/wp/getComment.php"
        	"tests/xmlrpc/wp/editTerm.php"
        	"tests/xmlrpc/wp/editProfile.php"
        	"tests/xmlrpc/wp/editPost.php"
        	"tests/xmlrpc/wp/deleteTerm.php"
        	"tests/xmlrpc/wp/deletePost.php"
        	"tests/xmlrpc/mw/newPost.php"
        	"tests/xmlrpc/mw/getRecentPosts.php"
        	"tests/xmlrpc/mw/getPost.php"
        	"tests/xmlrpc/mw/editPost.php"
        	"tests/xmlrpc/mt/getRecentPostTitles.php"
        	"tests/rest-api/rest-users-controller.php"
        	"tests/filesystem/findFolder.php"
        	"tests/admin/includesPlugin.php"
```

Comment (brandon-m-skinner) [inline bin/utils.sh:107]:
```suggestion
		"tests/term/wpDeleteTerm.php"
        	"tests/term/query.php"
        	"tests/term/getTerms.php"
        	"tests/xmlrpc/basic.php"
        	"tests/user/wpSendUserRequest.php"
        	"tests/user/capabilities.php"
        	"tests/term/wpGetObjectTerms.php"
        	"tests/term/getTheTerms.php"
        	"tests/term/getTerm.php"
        	"tests/term/cache.php"
        	"tests/term.php"
        	"tests/taxonomy.php"
        	"tests/sitemaps/sitemaps.php"
        	"tests/sitemaps/functions.php"
        	"tests/rest-api/rest-schema-setup.php"
        	"tests/rest-api/rest-posts-controller.php"
        	"tests/rest-api/rest-attachments-controller.php"
        	"tests/query/search.php"
        	"tests/query/isTerm.php"
        	"tests/privacy/wpPrivacySendRequestConfirmationNotification.php"
        	"tests/privacy/wpPrivacySendPersonalDataExportEmail.php"
        	"tests/privacy/wpPrivacySendErasureFulfillmentNotification.php"
        	"tests/post/getLastPostModified.php"
        	"tests/option/updateOption.php"
        	"tests/mail.php"
        	"tests/import/postmeta.php"
        	"tests/import/parser.php"
        	"tests/import/import.php"
        	"tests/filesystem/base.php"
        	"tests/feed/rss2.php"
        	"tests/dependencies/scripts.php"
        	"tests/date/getFeedBuildDate.php"
        	"tests/customize/custom-css-setting.php"
        	"tests/comment/wpCountComments.php"
        	"tests/canonical/sitemaps.php"
        	"tests/admin/includesScreen.php"
        	"tests/filesystem/base.php"
        	"tests/feed/atom.php"
        	"tests/adminbar.php"
        	"tests/xmlrpc/wp/restoreRevision.php"
        	"tests/xmlrpc/wp/newTerm.php"
        	"tests/xmlrpc/wp/newPost.php"
        	"tests/xmlrpc/wp/getUsers.php"
        	"tests/xmlrpc/wp/getUser.php"
        	"tests/xmlrpc/wp/getTerms.php"
        	"tests/xmlrpc/wp/getTerm.php"
        	"tests/xmlrpc/wp/getTaxonomy.php"
        	"tests/xmlrpc/wp/getTaxonomies.php"
        	"tests/xmlrpc/wp/getRevisions.php"
        	"tests/xmlrpc/wp/getProfile.php"
        	"tests/xmlrpc/wp/getPosts.php"
        	"tests/xmlrpc/wp/getPostTypes.php"
        	"tests/xmlrpc/wp/getPostType.php"
        	"tests/xmlrpc/wp/getPost.php"
        	"tests/xmlrpc/wp/getPages.php"
        	"tests/xmlrpc/wp/getPageList.php"
        	"tests/xmlrpc/wp/getPage.php"
        	"tests/xmlrpc/wp/getOptions.php"
        	"tests/xmlrpc/wp/getMediaItem.php"
        	"tests/xmlrpc/wp/getComments.php"
        	"tests/xmlrpc/wp/getComment.php"
        	"tests/xmlrpc/wp/editTerm.php"
        	"tests/xmlrpc/wp/editProfile.php"
        	"tests/xmlrpc/wp/editPost.php"
        	"tests/xmlrpc/wp/deleteTerm.php"
        	"tests/xmlrpc/wp/deletePost.php"
        	"tests/xmlrpc/mw/newPost.php"
        	"tests/xmlrpc/mw/getRecentPosts.php"
        	"tests/xmlrpc/mw/getPost.php"
        	"tests/xmlrpc/mw/editPost.php"
        	"tests/xmlrpc/mt/getRecentPostTitles.php"
        	"tests/rest-api/rest-users-controller.php"
        	"tests/filesystem/findFolder.php"
        	"tests/admin/includesPlugin.php"
```

**Comment by pschoffer** [comment]
Closing this PR as it is in a broken state. (@brandon-m-skinner reached the limits of Github, which is impressive! :))

New PR  - https://github.com/Automattic/vip-go-mu-plugins/pull/1791

**Pull request closed without merging**